service/history/api/multioperation/api.go

```diff
@@ -248,23 +268,10 @@ func startAndUpdateWorkflow(
 		return nil, serviceerror.NewUnavailable("Workflow could not be started as it is already running")
 	}
 
-	// without this, there's no Update registry on the call from Matching back to History
-	// TODO: eventually, we'll want to put the MS into the cache as well
-	workflowCache := workflowConsistencyChecker.GetWorkflowCache()
-	if _, err = workflowCache.Put(
-		shardContext,
-		namespaceID,
-		&commonpb.WorkflowExecution{WorkflowId: workflowKey.WorkflowID, RunId: workflowKey.RunID},
-		workflowCtx,
-		shardContext.GetMetricsHandler(),
-	); err != nil {
-		return nil, err
-	}
-
 	// wait for the update to complete
 	updateResp, err := updater.OnSuccess(ctx)
 	if err != nil {
-		return nil, serviceerror.NewUnavailable(fmt.Errorf("failed to complete Workflow Update: %w", err).Error())
+		return nil, newMultiOpError(nil, err) // `nil` for start since it succeeded
```

Comment:
Still not 100% sure what this should be; but this seems better than before. Maybe.

Reply:
It looks definitely better to me.